corehq/apps/receiverwrapper/views.py

```diff
@@ -81,6 +81,13 @@ def post(request, domain, app_id=None):
 
 
 def _noauth_post(request, domain, app_id=None):
+    """
+    This is explictly called for a submission that has secure submissions enabled, but is manually
+    overriding the submit URL to not specify auth context. It appears to be used by demo mode.
+
+    It mainly just checks that we are touching test data only in the right domain and submitting
+    as demo_user.
```

Comment:
I believe the mobile sets this. On device logs and demo_user submissions (I think just those two) it will pass in `authtype=noauth` as GET params to the url